lib/core/proxy.c

```diff
@@ -572,6 +573,12 @@ static int on_1xx(h2o_http1client_t *client, int minor_version, int status, h2o_
             h2o_push_path_in_link_header(self->src_req, headers[i].value.base, headers[i].value.len);
     }
 
+    if (status == 103) {
+        self->src_req->res.status = 103;
```

Comment:
`h2o_send_early_hints` is a function specific to handling 103. That would mean that it sends a set of header fields and not the status code itself as an argument.

Therefore, we should not set `h2o_req_t::res.status` to 103 or expect the value to be that.